e2e/test/iothub/messaging/MessageReceiveE2ETests.cs

```diff
@@ -501,32 +482,89 @@ private static async Task ReceiveMessageWithoutTimeoutCheckAsync(DeviceClient dc
             }
         }
 
-        private static async Task ReceiveMessageWithTimeoutCheckAsync(DeviceClient dc, TimeSpan timeout)
```

Comment:
This test only asserted that the transport handler returns the ReceiveAsync() call within the timeout specified. It did not actually send any c2d message via the service client, to test the e2e flow. For this reason, I modified this to be a unit test instead.